crates/services/txpool_v2/src/tests/pool.rs

```diff
@@ -0,0 +1,1256 @@
+    universe.config.black_list.coins.insert(utxo_id);
+    universe.build_pool();
+    let tx = universe.build_script_transaction(Some(vec![coin]), None, 0);
+
+    // When
+    let err = universe.verify_and_insert(tx).await.unwrap_err();
+
+    // Then
+    assert!(matches!(err, Error::BlacklistedUTXO(id) if id == utxo_id));
+}
+
+#[tokio::test]
+async fn insert__tx_with_blacklisted_owner() {
+    let mut universe = TestPoolUniverse::default();
+
+    // Given
+    let coin = universe.setup_coin().1;
+    let owner_addr = *coin.input_owner().unwrap();
+    universe.config.black_list.owners.insert(owner_addr);
+    universe.build_pool();
+    let tx = universe.build_script_transaction(Some(vec![coin]), None, 0);
+
+    // When
+    let err = universe.verify_and_insert(tx).await.unwrap_err();
+
+    // Then
+    assert!(matches!(err, Error::BlacklistedOwner(id) if id == owner_addr));
+}
+
+#[tokio::test]
+async fn insert__tx_with_blacklisted_contract() {
+    let mut universe = TestPoolUniverse::default();
+    let contract_id = Contract::EMPTY_CONTRACT_ID;
+
+    // Given
+    universe.config.black_list.contracts.insert(contract_id);
+    universe.build_pool();
+    let tx = universe.build_script_transaction(
+        Some(vec![create_contract_input(
+            Default::default(),
+            Default::default(),
+            contract_id,
+        )]),
+        Some(vec![Output::contract(
+            0,
+            Default::default(),
+            Default::default(),
+        )]),
+        0,
+    );
+
+    // When
+    let err = universe.verify_and_insert(tx).await.unwrap_err();
+
+    // Then
+    assert!(matches!(err, Error::BlacklistedContract(id) if id == contract_id));
+}
+
+#[tokio::test]
+async fn insert__tx_with_blacklisted_message() {
+    let mut universe = TestPoolUniverse::default();
+
+    // Given
+    let (message, input) = create_message_predicate_from_message(5000, 0);
+    let nonce = *message.nonce();
+    universe.config.black_list.messages.insert(nonce);
+    universe.build_pool();
+    let tx = universe.build_script_transaction(Some(vec![input]), None, 0);
+
+    // When
+    let err = universe.verify_and_insert(tx).await.unwrap_err();
+
+    // Then
+    assert!(matches!(err, Error::BlacklistedMessage(id) if id == nonce));
+}
+
+#[tokio::test]
+async fn insert_tx2_dependent_tx1() {
+    let mut universe = TestPoolUniverse::default();
+    universe.build_pool();
+
+    // Given
+    let (output, unset_input) = universe.create_output_and_input();
+    let tx1 = universe.build_script_transaction(None, Some(vec![output]), 0);
+
+    let input = unset_input.into_input(UtxoId::new(tx1.id(&ChainId::default()), 0));
+    let tx2 = universe.build_script_transaction(Some(vec![input]), None, 0);
+
+    // When
+    let result1 = universe.verify_and_insert(tx1).await;
+    let result2 = universe.verify_and_insert(tx2).await;
+
+    // Then
+    assert!(result1.is_ok());
+    assert!(result2.is_ok());
```

Comment:
There are a bunch of places where you `assert` the preliminary transactions' result.

I think it makes more sense to just unwrap the first result and move it into the `// Given` section. It's part of the setup.
```suggestion
    universe.verify_and_insert(tx1).await.unwrap();

    // When
    let result = universe.verify_and_insert(tx2).await;

    // Then
    assert!(result.is_ok());
```

There are a bunch of places like this. I'll avoid calling them all out to reduce noise. LMK if some places aren't obvious.

Reply:
Ok changed everywhere